### DISCUSSION TITLE
Rename files to post titles?

### DISCUSSION THREAD

**Discussion by mykeehu** [AAndyProgram/SCrawler#94] [Q&A] [answered]
I use the program to download from Reddit. 
In RipMe, there was an option to match the filename of the media with the name of the post (within reason, so the total length of the path will not be longer than 255 characters). Is there an option here to match the filename with the title of the post? (numbered for multiple images per post)

**Comment by AAndyProgram** [comment, marked as answer]
I'm not sure I fully understand what you want. If you want to name files by post name, then no, I don't plan to add this due to some factors. For example, the length of the path to the file. Also, the post may not have the title.

If your question is just to make sure SCrawler doesn't download a post that has already been downloaded, then no, SCrawler remembers the post IDs and other post information which prevents same post from being downloaded.

If you mean something else, please describe in more detail.

**Reply by mykeehu** [reply]
I was thinking that most posts would have a title, and when you downloaded the media for a post, the media file name would include the title of the post (say the first 100 characters), so it would be easier to later retrieve on Reddit which post it came from. Like We dived to the bottom of the ocean and stroked a shark.jpg

**Reply by AAndyProgram** [reply]
Okay, let's say Reddit is the only site (in SCrawler) that requires title text when creating a post. Twitter and Instagram have no title text at all. Porn sites files (xvideos, PornHub, XHamster) are named by the video clip name.
